python/lsst/ts/vent/controller/controller.py

```diff
@@ -0,0 +1,314 @@
+# This file is part of ts_atbuilding_vents
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import os
+from enum import IntEnum
+
+import megaind
+import pymodbus.client
+
+from . import config as cfg
+from . import vfd
+
+__all__ = ["Controller"]
+logger = logging.getLogger(__name__)
+
+
+class VentGateState(IntEnum):
+    CLOSED = 0
+    """
+    The vent is closed
+    """
+
+    PARTIALLY_OPEN = 1
+    """
+    The vent is neither open nor closed
+    """
+
+    OPEN = 2
+    """
+    The vent is open
+    """
+
+    FAULT = -1
+    """
+    The vent is both open and closed
+    """
+
+
+class Controller:
+    """A controller that commands the components associated with the AT dome
+    vents and fans. The code in this class is meant to run on the Raspberry Pi
+    described in
+    https://confluence.lsstcorp.org/display/~fritzm/Auxtel+Vent+Gate+Automation
+    """
+
+    def __init__(self):
+        """Constructor."""
+        logger.debug("__init__")
+
+        vfd_hostname = cfg.VFD_HOSTNAME
+        if "VFD_HOSTNAME" in os.environ:
+            vfd_hostname = os.environ["VFD_HOSTNAME"]
+
+        vfd_port = cfg.VFD_PORT
+        if "VFD_PORT" in os.environ:
+            vfd_port = int(os.environ["VFD_PORT"])
+
+        self.vfd_client = pymodbus.client.AsyncModbusTcpClient(vfd_hostname, port=vfd_port)
+        self.vfd_slave = cfg.VFD_SLAVE
+        self.default_fan_frequency = cfg.VFD_MAX_FREQ
+
+    async def connect(self) -> None:
+        """Connects to the VFD via modbus.
+
+        Raises
+        ======
+        ModbusException if the VFD is not available.
+        """
+        await self.vfd_client.connect()
+
+    async def get_vfd_manual_control(self) -> bool:
+        """Returns the VFD setting for manual or automatic (modbus) control.
+
+        Returns
+        =======
+        True if the VFD is configured to be controlled externally, or False if
```

Comment:
Please, check the [numpy doc style guide](https://numpydoc.readthedocs.io/en/latest/format.html) to see how to document return types etc.